### PR TITLE
ci: prevent merge of PRs with do-not-merge label

### PR DESCRIPTION
**Pull request by burgerdev**

### CODE REVIEW

**Comment by burgerdev** [comment]
* [with label](https://github.com/edgelesssys/contrast/actions/runs/11879587736?pr=1006)
* [without label](https://github.com/edgelesssys/contrast/actions/runs/11879594007?pr=1006)

File: .github/workflows/require_label.yml [hunk 1]
```diff
@@ -22,3 +22,13 @@ jobs:
...
+  do-not-merge:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'do not merge') }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: prevent merge
+        run: |
+          tee "${GITHUB_STEP_SUMMARY}" <<'EOF'
+          This PR is labeled as `do not merge`, remove the label to make this check pass.
+          EOF
+          exit 1
```
Comment (katexochen) [inline .github/workflows/require_label.yml:34]:
```suggestion
          echo "::error This PR is labeled as `do not merge`, remove the label to make this check pass."
          exit 1
```

usually github won't link to the summary of the job on failure, maybe add an error message to the log with error prefix.

Comment (burgerdev) [reply .github/workflows/require_label.yml:34]: Hm, I wanted to have the message both in the job logs as well as in the summary, but I think `::error` won't play nice there. Let me rewrite it.

**Comment by burgerdev** [comment]
I'm not going to rerun the e2e tests unless asked to, the PR should be fine now.

**Review by katexochen — APPROVED** [review]
actually, I don't think we even need to write to summary, annotation is already shown there anyway:
image

But I'm also fine to merge like this.

**Comment by burgerdev** [comment]
Thanks, we don't need the summary then.

**Review by katexochen — APPROVED** [review]

**Pull request merged**